doc/specifications/translations/removal.md

```diff
@@ -14,7 +14,7 @@ the existing Versions of a Content Object and is specified as:
  *       a warning before performing it.
  *
  * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException if the specified translation
- *         is the only one Content Object has.
+ *         is the only one Content Object has or it is the main translation.
```

Comment:
> Removing main location is disallowed, but there is api to change main so this won't be a blocker right?

Exactly. To remove translation which is the main one, you have to update `mainLanguageCode`  content metadata. I should probably add some integration test proving that... ;)

> How do we handle initial language for versions? As it can be different one from version to version we will need to pick another one being initial version.

> On the initial version I might be wrong, but afaik the current proposed logic means it won't allow you to delete any language, as typically there will be one version for each translation added where that translation is the initial one.

I actually haven't seen this case when creating tests, but I would imagine it's quite probable.
My thought on this is then - in such case, let's update `initialLanguageCode` with `mainLanguageCode`.

> We will also need to clarify what we do with drafts, they typically only contain one language, so either removing the translation involves removing the draft which I guess should throw.

I wouldn't complicate removal API. I'd rather state in the doc, that you have to remove manually any version that has only one language.

> with all the different cases of throws here I guess we should model this better to clearly tell what is wrong.

On that I would need some help as I'm not sure what you're aiming here for. Ideally this probably should be different exception, but `BadStateException` really fits both cases. I ensured it throws different message. However, as I stated in previous comment, exception related to a translation being the only one shouldn't happen now, as the one about `mainLanguageCode` happens sooner.
So, maybe I'll just drop "only one translation" check/exception?